astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
```

Comment:
Doing a file I/O 4x when 2x is sufficient is undesirable. I recommend you and @saimn sort out the misunderstanding before merge. Thanks!